[ci] use lintrunner in CI

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #68460

This changes our lint workflows to use lintrunner for the linters that
are currently supported

+ some random fixes to make things lint clean on master
+ changes to Makefile to use lintrunner

Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

suo committed:
test lint

[ghstack-poisoned]

suo commented:
@suo has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D32470350).

suo committed:
Update on "test lint"


Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "test lint"


Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "test lint"


Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "test lint"


Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "test lint"


Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "test lint"


Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "test lint"


Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

t10-13rocket reviewed (approved):
LGTM

suo committed:
Update on "test lint"


Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "test lint"


Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "test lint"


Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "test lint"


Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "test lint"


Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "test lint"


Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "test lint"


Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "test lint"


Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "test lint"


Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "test lint"


Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "test lint"


Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "test lint"


Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "test lint"


Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "test lint"


Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "test lint"


Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "test lint"


Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "test lint"


Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "test lint"


Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "test lint"


Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "test lint"


Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "test lint"


Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "test lint"


Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "test lint"


Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "test lint"


Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "test lint"


Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "test lint"


Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "[ci] use lintrunner in CI"


This changes our lint workflows to use lintrunner for the linters that
are currently supported

+ some random fixes to make things lint clean on master
+ changes to Makefile to use lintrunner

Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "[ci] use lintrunner in CI"


This changes our lint workflows to use lintrunner for the linters that
are currently supported

+ some random fixes to make things lint clean on master
+ changes to Makefile to use lintrunner

Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "[ci] use lintrunner in CI"


This changes our lint workflows to use lintrunner for the linters that
are currently supported

+ some random fixes to make things lint clean on master
+ changes to Makefile to use lintrunner

Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "[ci] use lintrunner in CI"


This changes our lint workflows to use lintrunner for the linters that
are currently supported

+ some random fixes to make things lint clean on master
+ changes to Makefile to use lintrunner

Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "[ci] use lintrunner in CI"


This changes our lint workflows to use lintrunner for the linters that
are currently supported

+ some random fixes to make things lint clean on master
+ changes to Makefile to use lintrunner

Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "[ci] use lintrunner in CI"


This changes our lint workflows to use lintrunner for the linters that
are currently supported

+ some random fixes to make things lint clean on master
+ changes to Makefile to use lintrunner

Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "[ci] use lintrunner in CI"


This changes our lint workflows to use lintrunner for the linters that
are currently supported

+ some random fixes to make things lint clean on master
+ changes to Makefile to use lintrunner

Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "[ci] use lintrunner in CI"


This changes our lint workflows to use lintrunner for the linters that
are currently supported

+ some random fixes to make things lint clean on master
+ changes to Makefile to use lintrunner

Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "[ci] use lintrunner in CI"


This changes our lint workflows to use lintrunner for the linters that
are currently supported

+ some random fixes to make things lint clean on master
+ changes to Makefile to use lintrunner

Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "[ci] use lintrunner in CI"


This changes our lint workflows to use lintrunner for the linters that
are currently supported

+ some random fixes to make things lint clean on master
+ changes to Makefile to use lintrunner

Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

seemethere reviewed (approved):
w00t!

suo committed:
Update on "[ci] use lintrunner in CI"


This changes our lint workflows to use lintrunner for the linters that
are currently supported

+ some random fixes to make things lint clean on master
+ changes to Makefile to use lintrunner

Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "[ci] use lintrunner in CI"


This changes our lint workflows to use lintrunner for the linters that
are currently supported

+ some random fixes to make things lint clean on master
+ changes to Makefile to use lintrunner

Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "[ci] use lintrunner in CI"


This changes our lint workflows to use lintrunner for the linters that
are currently supported

+ some random fixes to make things lint clean on master
+ changes to Makefile to use lintrunner

Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo committed:
Update on "[ci] use lintrunner in CI"


This changes our lint workflows to use lintrunner for the linters that
are currently supported

+ some random fixes to make things lint clean on master
+ changes to Makefile to use lintrunner

Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

janeyx99 commented on Makefile:
```diff
@@ -51,12 +52,7 @@ shellcheck:
-		--job 'cmakelint' --step 'Install dependencies' --no-quiet
-	$(PYTHON) tools/actions_local_runner.py --file .github/workflows/lint.yml \
-		--job 'mypy' --step 'Install dependencies' --no-quiet
+	$(PIP) install lintrunner
```
soooo clean

suo commented:
@pytorchbot merge this

malfet commented:
@pytorchbot revert this to fix merge conflict with  https://github.com/pytorch/pytorch/commit/6581afe58afae4dcc34d4024499c6cb61a56b448 (will reland without the conflict later)

suo committed:
Update on "[ci] use lintrunner in CI"


This changes our lint workflows to use lintrunner for the linters that
are currently supported

+ some random fixes to make things lint clean on master
+ changes to Makefile to use lintrunner

Differential Revision: [D32470350](https://our.internmc.facebook.com/intern/diff/D32470350)

[ghstack-poisoned]

suo commented:
@pytorchbot merge this

suo commented:
@pytorchbot merge this
